content/frontend/react-apollo/9-pagination.md

```diff
@@ -102,261 +158,349 @@ export const FEED_QUERY = gql`
+then reload the component with a new `page` in the URL that
+will be used to calculate the right chunk of links to load.
 
-The implementation of these is very simple. You're retrieving the current page from the url and implement a sanity check to make sure that it makes sense to paginate back or forth. Then you simply calculate the next page and tell the router where to navigate next. The router will then reload the component with a new `page` in the url that will be used to calculate the right chunk of links to load. Run the app by typing `yarn start` in a terminal and use the new buttons to paginate through your list of links!
+Run the app by typing `yarn start` in a terminal and use the
+new buttons to paginate through the list of links!
 
-### Final adjustments
+### Final Adjustments
 
-Through the changes that we made to the `FEED_QUERY`, you'll notice that the `update` functions of your mutations don't work any more. That's because `readQuery` now also expects to get passed the same variables that we defined before.
+Through the changes that we made to the `FEED_QUERY`, we'll
+notice that the `update` functions of the mutations don't
+work any more. That's because `readQuery` now also expects
+to get passed the same variables that we defined before.
 
-> **Note**: `readQuery` essentially works in the same way as the `query` method on the `ApolloClient` that you used to implement the search. However, instead of making a call to the server, it will simply resolve the query against the local store! If a query was fetched from the server with variables, `readQuery` also needs to know the variables to make sure it can deliver the right information from the cache.
+> **Note**: `readQuery` essentially works in the same way as
+> the `query` method on the `ApolloClient` that we used to
+> implement the search. However, instead of making a call to
+> the server, it will simply resolve the query against the
+> local store! If a query was fetched from the server with
+> variables, `readQuery` also needs to know the variables to
+> make sure it can deliver the right information from the
+> cache.
 
 <Instruction>
 
-With that information, open `LinkList.js` and update the `_updateCacheAfterVote` method to look as follows:
-
-```js{2-11}(path=".../hackernews-react-apollo/src/components/LinkList.js")
-_updateCacheAfterVote = (store, createVote, linkId) => {
-  const isNewPage = this.props.location.pathname.includes('new')
-  const page = parseInt(this.props.match.params.page, 10)
+With that information, open `Link.js` and update the
+`update` function on the `useMutation` hook:
 
-  const skip = isNewPage ? (page - 1) * LINKS_PER_PAGE : 0
-  const first = isNewPage ? LINKS_PER_PAGE : 100
-  const orderBy = isNewPage ? 'createdAt_DESC' : null
-  const data = store.readQuery({
-    query: FEED_QUERY,
-    variables: { first, skip, orderBy }
-  })
+```js{2-11}(path=".../hackernews-react-apollo/src/components/Link.js")
```

Comment:
Let's update the diffs here 🙂
```suggestion
```js{5-7, 16-20, 40-44}(path=".../hackernews-react-apollo/src/components/Link.js")
```